src/encoded_core/types/file.py

```diff
@@ -298,15 +298,15 @@ def _update(self, properties, sheets=None):
                 xfile['accession'] = properties.get('accession')
                 # just need a filename to trigger creation of credentials
                 xfile_name = xfile.get("filename")
-                if xfile_name is None:
-                    xfile['filename'] = xfile['accession']
+                file_extension = xfile_format.properties.get('standard_file_extension')
+                if xfile_name is None:  # assume if its already there its correct
+                    xfile['filename'] = '{}.{}'.format(xfile['accession'], file_extension)
                 xfile['uuid'] = str(uuid)
                 # if not 'status' in xfile or not bool(xfile['status']):
                 #    xfile['status'] = properties.get('status')
                 ext = self.build_external_creds(self.registry, uuid, xfile)
                 # build href
-                file_extension = xfile_format.properties.get('standard_file_extension')
-                filename = '{}.{}'.format(xfile['accession'], file_extension)
+                filename = xfile['filename']
```

Comment:
This should be reverted to the line above, otherwise the `href` is going to be using the annotated filename for the location in S3.
```suggestion
                filename = '{}.{}'.format(xfile['accession'], file_extension)
```